### PR TITLE
feat: when writing delta lake tables, include a shallow FHIR schema

### PR DESCRIPTION
**Pull request by mikix**
This helps downstream SQL not have to worry about whether the incoming data has a particular optional FHIR field (which before, would mean that the Athena schema didn't define it, which would yield a SQL error).

Now, all the R4 FHIR fields will be represented at least at the toplevel, and SQL can rest a bit easier.

### Checklist
- [x] Consider if documentation (like in `docs/`) needs to be updated
- [x] Consider if tests should be added

### CODE REVIEW

File: cumulus_etl/chart_review/cli.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import ctakesclient
 from ctakesclient.typesystem import Polarity
 
-from cumulus_etl import cli_utils, common, deid, errors, fhir_client, fhir_common, loaders, nlp, store
+from cumulus_etl import cli_utils, common, deid, errors, fhir, loaders, nlp, store
```
Comment (mikix) [inline cumulus_etl/chart_review/cli.py:11]: You're going to see a lot of light refactoring. I was adding a new "FHIR spec"-like bit of code, so I took the opportunity to gather the FHIR code into one subdirectory.

`fhir_client` -> `fhir`
`fhir_common` -> `fhir`
(new) `fhir_schemas` -> `fhir`

And I moved the general purpose `FatalError` out of `fhir_client` and into `errors`.

File: pyproject.toml [hunk 2]
```diff
@@ -13,14 +13,15 @@ requires-python = ">= 3.10"
 dependencies = [
     "ctakesclient >= 3, < 4",
     "delta-spark >= 2.3, < 3",
+    "fhirclient < 5",
```
Comment (mikix) [inline pyproject.toml:16]: I know we just got rid of `fhirclient`, but it was the quickest/easiest way I could find to get python-friendly flattened versions of the FHIR spec.

File: pyproject.toml [hunk 3]
```diff
@@ -13,14 +13,15 @@ requires-python = ">= 3.10"
...
     "httpx < 1",
     "inscriptis < 3",
     "jwcrypto < 2",
     "label-studio-sdk < 1",
     "oracledb < 2",
     "pandas < 3",
     "philter-lite < 1",
-    "pyarrow < 12",
+    "pyarrow < 13",
```
Comment (mikix) [inline pyproject.toml:24]: Not necessary, I just noticed we were a version back while I was playing with data transforms.

File: cumulus_etl/formats/deltalake.py [hunk 4]
```diff
@@ -165,3 +157,45 @@ def _configure_fs(root: store.Root, spark: pyspark.sql.SparkSession):
         region_name = fsspec_options.get("client_kwargs", {}).get("region_name")
         if region_name:
             spark.conf.set("fs.s3a.endpoint.region", region_name)
+
+    @contextlib.contextmanager
+    def pandas_to_spark_with_schema(self, dataframe: pandas.DataFrame) -> pyspark.sql.DataFrame:
+        """Transforms a pandas DF to a spark DF with a full FHIR schema included"""
+        # This method solves two problems:
+        # 1. Pandas schemas are very loosey-goosey (and don't really take nested data into account), so simply
```
Comment (dogversioning) [inline cumulus_etl/formats/deltalake.py:165]: such accurate spelling of loosey-goosey!

File: cumulus_etl/chart_review/cli.py (hunk 1 above)
Comment (dogversioning) [reply cumulus_etl/chart_review/cli.py:11]: this generally feels like a very good move to me.

**Review by dogversioning — APPROVED** [review]

**Comment by mikix** [comment]
Tested this on some BCH delta lakes and it seemed to work & fix our issues with the `core` library. Merging away.

**Pull request merged**